File: 2025-09-23 - Pilhas/Exemplo_Python/TestePilha.py
# ------------------------------------------------------------------------
# TestePilha.py
# Programa principal que demonstra todas as operações básicas da pilha.
# Usa a variável pilha_disciplinas e as disciplinas fornecidas.
# ------------------------------------------------------------------------
from Node import Node
from Pilha import Pilha

def main():
   # Cria a pilha
   pilhaDisciplinas = Pilha()

   # Inserções
   print("Inserindo elementos na pilha...")
   pilhaDisciplinas.push(Node("TEC.0007", "Programação de Computadores", 120))
   pilhaDisciplinas.push(Node("TEC.0027", "Estrutura de Dados Não-Lineares", 80))
   pilhaDisciplinas.push(Node("TEC.0142", "Programação para Redes", 120))
   pilhaDisciplinas.push(Node("TEC.0022", "Bancos de Dados", 80))
   pilhaDisciplinas.imprimirLista()
   print("Tamanho após inserções:", pilhaDisciplinas.tamanho(), "\n")

   # Peek
   print("Peek (topo atual):", pilhaDisciplinas.peek())
   print("Tamanho permanece:", pilhaDisciplinas.tamanho(), "\n")

   # Pop
   print("Removendo (pop) elemento do topo...")
   removido = pilhaDisciplinas.pop()
   print("Elemento removido:", removido)
   print("Tamanho após pop:", pilhaDisciplinas.tamanho(), "\n")

   # Busca por código
   codigo_busca = "TEC.0027"
   print(f"Buscando código {codigo_busca}...")
   encontrado = pilhaDisciplinas.buscarPorCodigo(codigo_busca)
   print("Encontrado:" if encontrado else "Não encontrado:", encontrado, "\n")

   # Atualização
   print("Atualizando TEC.0027 (novo nome e nova carga)...")
   atualizado = pilhaDisciplinas.atualizar(
      "TEC.0027",
      "Estrutura de Dados Não-Lineares - Avançado",
      96
   )
   print("Atualizado com sucesso?", atualizado)
   pilhaDisciplinas.imprimirLista()
   print("Tamanho:", pilhaDisciplinas.tamanho(), "\n")

   # Remoção por código
   print("Removendo por código TEC.0007...")
   removido_codigo = pilhaDisciplinas.removerPorCodigo("TEC.0007")
   print("Removido TEC.0007?", removido_codigo)
   pilhaDisciplinas.imprimirLista()
   print("Tamanho:", pilhaDisciplinas.tamanho(), "\n")

   # Verificação contains
   print("Verificando se existe TEC.0022:", pilhaDisciplinas.contains("TEC.0022"))
   print("Verificando se existe TEC.0007:", pilhaDisciplinas.contains("TEC.0007"), "\n")

   # Limpar pilha
   print("Esvaziando a pilha...")
   pilhaDisciplinas.clear()
   print("Pilha vazia?", pilhaDisciplinas.isEmpty())
   print("Tamanho final:", pilhaDisciplinas.tamanho())

if __name__ == "__main__":
   main()

File: 2025-09-23 - Pilhas/Exemplo_C#/TestePilha.cs
// ------------------------------------------------------------------------
// TestePilha.cs
// Programa principal que demonstra todas as operações da pilha.
// Usa a variável pilhaDisciplinas e as disciplinas fornecidas.
// ------------------------------------------------------------------------
using System;

public class TestePilha
{
   public static void Main(string[] args)
   {
      // ------------------------------------------------------------------
      // Cria a pilha
      // ------------------------------------------------------------------
      Pilha pilhaDisciplinas = new Pilha();

      // ------------------------------------------------------------------
      // Inserções (push)
      // ------------------------------------------------------------------
      Console.WriteLine("Inserindo elementos na pilha...");
      pilhaDisciplinas.Push(new Node("TEC.0007", "Programação de Computadores", 120));
      pilhaDisciplinas.Push(new Node("TEC.0027", "Estrutura de Dados Não-Lineares", 80));
      pilhaDisciplinas.Push(new Node("TEC.0142", "Programação para Redes", 120));
      pilhaDisciplinas.Push(new Node("TEC.0022", "Bancos de Dados", 80));
      pilhaDisciplinas.ImprimirLista();
      Console.WriteLine("Tamanho após inserções: " + pilhaDisciplinas.Tamanho());
      Console.WriteLine();

      // Peek
      Console.WriteLine("Peek (topo atual): " + pilhaDisciplinas.Peek());
      Console.WriteLine("Tamanho permanece: " + pilhaDisciplinas.Tamanho());
      Console.WriteLine();

      // Pop
      Console.WriteLine("Removendo (pop) elemento do topo...");
      Node removido = pilhaDisciplinas.Pop();
      Console.WriteLine("Elemento removido: " + removido);
      Console.WriteLine("Tamanho após pop: " + pilhaDisciplinas.Tamanho());
      Console.WriteLine();

      // Busca por código
      string codigoBusca = "TEC.0027";
      Console.WriteLine("Buscando código " + codigoBusca + "...");
      Node encontrado = pilhaDisciplinas.BuscarPorCodigo(codigoBusca);
      Console.WriteLine(encontrado != null
         ? "Encontrado: " + encontrado
         : "Código " + codigoBusca + " não encontrado.");
      Console.WriteLine();

      // Atualização
      Console.WriteLine("Atualizando TEC.0027 (novo nome e nova carga)...");
      bool atualizado = pilhaDisciplinas.Atualizar(
         "TEC.0027",
         "Estrutura de Dados Não-Lineares - Avançado",
         96
      );
      Console.WriteLine("Atualizado com sucesso? " + atualizado);
      pilhaDisciplinas.ImprimirLista();
      Console.WriteLine("Tamanho: " + pilhaDisciplinas.Tamanho());
      Console.WriteLine();

      // Remoção por código
      Console.WriteLine("Removendo por código TEC.0007...");
      bool removidoPorCodigo = pilhaDisciplinas.RemoverPorCodigo("TEC.0007");
      Console.WriteLine("Removido TEC.0007? " + removidoPorCodigo);
      pilhaDisciplinas.ImprimirLista();
      Console.WriteLine("Tamanho: " + pilhaDisciplinas.Tamanho());
      Console.WriteLine();

      // Contains
      Console.WriteLine("Verificando se existe TEC.0022: " + pilhaDisciplinas.Contains("TEC.0022"));
      Console.WriteLine("Verificando se existe TEC.0007: " + pilhaDisciplinas.Contains("TEC.0007"));
      Console.WriteLine();

      // Clear
      Console.WriteLine("Esvaziando a pilha...");
      pilhaDisciplinas.Clear();
      Console.WriteLine("Pilha vazia? " + pilhaDisciplinas.IsEmpty());
      Console.WriteLine("Tamanho final: " + pilhaDisciplinas.Tamanho());
   }
}
